src/apollo/queries.js

```diff
@@ -1,5 +1,124 @@
 import gql from 'graphql-tag';
 
+export const UNISWAP_PAIRS_ID_QUERY = gql`
+  query pairs {
+    pairs(first: 200, orderBy: trackedReserveETH, orderDirection: desc) {
+      id
+    }
+  }
+`;
+
+export const UNISWAP_PAIR_DATA_QUERY = (pairAddress, block) => {
+  const queryString = `
+    fragment PairFields on Pair {
+      id
+      txCount
+      token0 {
+        id
+        symbol
+        name
+        totalLiquidity
+        derivedETH
+      }
+      token1 {
+        id
+        symbol
+        name
+        totalLiquidity
+        derivedETH
+      }
+      reserve0
+      reserve1
+      reserveUSD
+      totalSupply
+      trackedReserveETH
+      reserveETH
+      volumeUSD
+      untrackedVolumeUSD
+      token0Price
+      token1Price
+      createdAtTimestamp
+  }
+  query pairs {
+    pairs(${
+      block ? `block: {number: ${block}}` : ``
+    } where: { id: "${pairAddress}"} ) {
+      ...PairFields
+    }
+  }`;
+  return gql`
+    ${queryString}
+  `;
+};
+
+export const UNISWAP_PAIRS_BULK_QUERY = gql`
```

Comment:
Same with this query (same fields as above comment)

Reply:
Done!